security-analysis/security-analysis-default/src/main/java/com/powsybl/security/impl/DefaultSecurityAnalysis.java

```diff
@@ -296,16 +297,18 @@ private void checkPreContingencyViolations(Network network, Consumer<LimitViolat
         if (violationDetector != null) {
             violationDetector.checkAll(network, consumer);
         } else {
-            LimitViolationDetection.checkAll(network, EnumSet.allOf(LoadingLimitType.class), 1., consumer);
+            LimitViolationDetection.checkAll(network, EnumSet.allOf(LoadingLimitType.class), new LimitsComputer.NoModificationsImpl(), consumer);
```

Comment:
Use the singleton here:

```suggestion
            LimitViolationDetection.checkAll(network, EnumSet.allOf(LoadingLimitType.class), LimitsComputer.NO_MODIFICATIONS, consumer);
```